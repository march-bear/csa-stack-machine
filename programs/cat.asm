loop:
    input
    jz end
    print
    jmp loop
end:
    halt