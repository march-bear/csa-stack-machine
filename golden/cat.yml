in_source: |-
  loop:
    input
    jz end
    print
    jmp loop
  end:
    halt
in_stdin: |-
  foo
out_code: |-
  [{"opcode": "input", "term": [1, 4, "input"]},
   {"opcode": "jz", "term": [2, 4, "jz end"], "arg": 4},
   {"opcode": "print", "term": [3, 4, "print"]},
   {"opcode": "jmp", "term": [4, 4, "jmp loop"], "arg": 0},
   {"opcode": "halt", "term": [6, 4, "halt"]}]
out_stdout: |
  LoC: 7 code instr: 5
  ============================================================
  output: foo
  instr_counter: 14 ticks: 35
out_log: |